### PR TITLE
PPV2 fix updating name for all shops while enabling status for single shop

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | https://github.com/PrestaShop/PrestaShop/issues/31000
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | https://github.com/PrestaShop/PrestaShop/issues/31000
| Fixed ticket?     | https://github.com/PrestaShop/PrestaShop/issues/31000
| Related PRs       | 
| Sponsor company   |

### CODE REVIEW

File: src/Core/Form/IdentifiableObject/CommandBuilder/Product/UpdateProductCommandsBuilder.php [hunk 1]
```diff
@@ -70,12 +70,14 @@ public function buildCommands(ProductId $productId, array $formData, ShopConstra
             ->configureStockInformation($config, $formData)
         ;
 
-        $config->addField('[header][active]', 'setActive', DataField::TYPE_BOOL);
+        $config->addMultiShopField('[header][active]', 'setActive', DataField::TYPE_BOOL);
```
Comment (zuk3975) [inline src/Core/Form/IdentifiableObject/CommandBuilder/Product/UpdateProductCommandsBuilder.php:73]: active field was also forgotten to be handled for multiple shops here

File: src/Core/Form/IdentifiableObject/CommandBuilder/Product/UpdateProductCommandsBuilder.php [hunk 2]
```diff
@@ -239,4 +241,32 @@ private function configureStockInformation(CommandBuilderConfig $config, array $
...
+     * Name and status are related - when name is not filled, then product cannot be enabled.
+     * When name is being updated for all shops, but status only for single shop, then status would be filled into
+     * single shop command while name only in all shops command. Since single shop command is always executed first,
+     * it will try to enable product before name is inserted in allShopsCommand and will end up throwing
+     * error about name being empty.
+     *
+     * So to solve that, we check if status is being updated for single shop,
+     * and if that is the case, then we manually fill the name into single shop command too.
+     * So at the end it will end up updating name twice - with single shop command and with all shops command,
+     * but at least it won't throw the error.
```
Comment (zuk3975) [inline src/Core/Form/IdentifiableObject/CommandBuilder/Product/UpdateProductCommandsBuilder.php:255]: should be clear by this explanation

Comment (jolelievre) [reply src/Core/Form/IdentifiableObject/CommandBuilder/Product/UpdateProductCommandsBuilder.php:255]: Seems perfect!

File: src/Core/Form/IdentifiableObject/CommandBuilder/Product/UpdateProductCommandsBuilder.php (hunk 1 above)
Comment (jolelievre) [reply src/Core/Form/IdentifiableObject/CommandBuilder/Product/UpdateProductCommandsBuilder.php:73]: Nice catch!

**Review by jolelievre — APPROVED** [review]
Thanks @zuk3975

**Review by matthieu-rolland — APPROVED** [review]

**Review by florine2623 — COMMENTED** [review]
Hello @zuk3975 ,

Link to automated tests : https://github.com/florine2623/testing_pr/actions/runs/4122559456

The error message `To put this product online, please enter a name.` is well removed once the Name of product is filled.

https://user-images.githubusercontent.com/16019289/217548165-928bd4d3-beb4-4b16-85a1-91cb3ae08edf.mov

I have 2 different error messages for 2 different cases :
`To put this product online, please enter a name.`
AND
`The field name is required at least in your default language.`

As discussed with @zuk3975 , it is the expected behavior.
It's all based on this simple rule :
Product without name cannot be enabled (first message)
Once the product name is filled, you shouldn't be able to remove it anymore (second message)

It is QA ✅ !
Thanks 💪

**Review by florine2623 — APPROVED** [review]
QA ✅

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thank you @zuk3975

**Pull request merged**